pkg/trtl/honu.go

```diff
@@ -30,72 +30,68 @@ func NewHonuService(s *Server) (*HonuService, error) {
 }
 
 // Get is a unary request to retrieve a value for a key.
+// If metadata is requested in the GetRequest, the request will use honu.Object() to
+// retrieve the entire object, including the metadata. If no metadata is requested, the
+// request will use honu.Get() to get only the value.
+// If a namespace is provided, the namespace is passed to the internal honu Options,
+// to look in that namespace only.
 func (h *HonuService) Get(ctx context.Context, in *pb.GetRequest) (*pb.GetReply, error) {
 	var err error
 
 	if _, found := reservedNamespaces[in.Namespace]; found {
-		log.Warn().Msg("cannot use reserved namespace")
+		log.Warn().Str("namespace", in.Namespace).Msg("cannot use reserved namespace")
 		return nil, status.Error(codes.PermissionDenied, "cannot use reserved namespace")
 	}
 	if len(in.Key) == 0 {
 		log.Warn().Msg("missing key in Trtl Get request")
 		return nil, status.Error(codes.InvalidArgument, "key must be provided in Get request")
 	}
 
-	var key []byte
-	if len(in.Namespace) > 0 {
-		key = prepend(in.Namespace, in.Key)
-	} else {
-		key = prepend("default", in.Key)
-	}
-
-	if in.Options != nil {
-		log.Debug().Str("key", string(key)).Bool("return_meta", in.Options.ReturnMeta).Msg("Trtl Get")
-	} else {
-		log.Debug().Str("key", string(key)).Msg("Trtl Get")
-	}
 	if in.Options != nil && in.Options.ReturnMeta {
-		// Retrieve and return the metadata.
+		// Retrieve and return the metadata (uses honu.Object())
+		log.Debug().Str("key", string(in.Key)).Bool("return_meta", in.Options.ReturnMeta).Msg("Trtl Get")
+
+		// Check if we have a namespace
 		var object *object.Object
-		if object, err = h.db.Object(key); err != nil {
+		if in.Namespace != "" {
+			object, err = h.db.Object(in.Key, options.WithNamespace(in.Namespace))
+		} else {
+			object, err = h.db.Object(in.Key)
+		}
+
+		if err != nil {
 			// TODO: Check for the honu not found error instead.
```

Comment:
I think this should be unblocked now because the newest version of Honu should expose the NotFound error?